### DISCUSSION TITLE
oauth redirection from react client results in CORS issue

### DISCUSSION THREAD

**Discussion by saptha** [directus/directus#9979] [Archive]
From my React Client, I call /auth/login/google. and /auth/login/facebook
It fails with "Access to XMLHttpRequest at ... has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
The APIs work fine from Postman client.

**Comment by NilsBaumgartner1994** [comment]
Had the same problem:
You may encouter this problem if your react app is not running on the same host. This problem will stay and can't be changed:
 https://github.com/directus/directus/issues/8378 https://github.com/directus/directus/issues/8392 https://github.com/directus/directus/issues/9410

I have setup a solution/workaround for this to work:

https://gist.github.com/NilsBaumgartner1994/6e1a764491c736d7f187017c9e1f17cc

**Comment by NilsBaumgartner1994** [comment]
Be sure to handle the Token then correct

**Comment by rijkvanzanten** [comment]
How are you calling those endpoints from your React app? What browser are you using? Are your app / Directus hosted somewhere, or running locally? Are you using HTTPS on these deployments? What is the configuration for [the CORS settings in Directus](https://docs.directus.io/configuration/config-options/#cors)?

**Reply by saptha** [reply]
@rijkvanzanten, 
Using axios in React, I call the directus APIs hosted in a different subdomain or server. All are https enabled. I'm using default values for the CORS env variables. You can check the app at https://www.magicauthor.com/login

**Reply by rijkvanzanten** [reply]
Ahh I see. You're making a GET request to that endpoint. That particular endpoint is supposed to redirect you to the external provider to login. Try using a `<a href="https://prdapi.magicauthor.com/auth/login/facebook?redirect=https://www.magicauthor.com/login">` to have the user open that actual route on the page 🙂. The `redirect` query parameter will tell Directus to redirect back to your app on successful login.

**Reply by saptha** [reply]
I need the access token as well at the client side. I think I need to implement the workaround provided by @NilsBaumgartner1994 above.
Will try this

Thanks to you both.

**Reply by rijkvanzanten** [reply]
@saptha Once you're logged in, the Directus API will set a cookie containing the refresh token. With this cookie being set, you should be able to use the `/auth/refresh` endpoint to obtain the access token 👍🏻

**Reply by saptha** [reply]
Oh ok. Will keep you posted with the results.
Thanks Rijk

**Reply by NilsBaumgartner1994** [reply]
@saptha you can checkout this example application with single code base for a web and native application:
https://github.com/FireboltCasters/Raiders-of-Storm

**Reply by saptha** [reply]
@NilsBaumgartner1994 Thanks for the example code base
Thanks @rijkvanzanten 
I've implemented this logic and social logins are now working fine.
